google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb

```diff
@@ -351,6 +352,137 @@ def expires_in= ttl
           @resource_name = nil
         end
 
+        ##
+        # Returns the {Topic} to which dead letter messages should be published if a dead letter policy is configured,
+        # otherwise `nil`. Dead lettering is done on a best effort basis. The same message might be dead lettered
+        # multiple times.
+        #
+        # See also {#dead_letter_topic=}, {#dead_letter_max_delivery_attempts=} and
+        # {#dead_letter_max_delivery_attempts}.
+        #
+        # Makes an API call to retrieve the topic name when called on a reference object. See {#reference?}.
+        #
+        # @return [Topic, nil]
+        #
+        # @example
+        #   require "google/cloud/pubsub"
+        #
+        #   pubsub = Google::Cloud::PubSub.new
+        #
+        #   sub = pubsub.subscription "my-topic-sub"
+        #   sub.dead_letter_topic.name #=> "projects/my-project/topics/my-dead-letter-topic"
+        #   sub.dead_letter_max_delivery_attempts #=> 10
+        #
+        def dead_letter_topic
```

Comment:
I don think we are exposing this as an API in other languages for cloud pubsub. Can you confirm please? We should keep the APIs consistent.

Reply:
How is it done in other languages, as a `DeadLetterPolicy` type? The style of the Ruby client libraries is typically to flatten nested fields.

Reply:
In Go, we created a type for `DeadLetterPolicy` but I think libraries should remain idiomatic with respect to the language rather than across libraries.

Reply:
@hongalex In that case we should keep the flattened design here, since the style is already long established in the google-cloud-ruby client libraries, and in my opinion seems idiomatic to Ruby users.

Reply:
SGTM